include/mbgl/renderer/render_target.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+
+namespace mbgl {
+
+namespace gfx {
+class Texture2D;
+class UploadPass;
+using Texture2DPtr = std::shared_ptr<Texture2D>;
+} // namespace gfx
+
+class LayerGroupBase;
+class PaintParameters;
+class RenderOrchestrator;
+class RenderTree;
+
+using LayerGroupBasePtr = std::shared_ptr<LayerGroupBase>;
+
+class RenderTarget {
+public:
+    virtual ~RenderTarget() = default;
+
+    void setTexture(const gfx::Texture2DPtr& texture_) { texture = std::move(texture_); };
+    const gfx::Texture2DPtr& getTexture() const { return texture; };
+
+    bool addLayerGroup(LayerGroupBasePtr, bool replace);
+    bool removeLayerGroup(const int32_t layerIndex);
+    size_t numLayerGroups() const noexcept;
+    const LayerGroupBasePtr& getLayerGroup(const int32_t layerIndex) const;
+    void observeLayerGroups(std::function<void(LayerGroupBase&)>);
```

Comment:
I see lots of functions introduced with the prefix "observe" while effectively what it is doing is visiting? Was such naming agreed upon? If yes I wonder what is the idea behind such naming.

Reply:
No specific reasoning, just used the wrong word, I think.  I have no objection to changing it globally.

Reply:
That seems reasonable.